ScatterPeas/scatterpeas/tests/test_views.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import pytest
+import os
+import datetime
+from sqlalchemy import create_engine
+from scatterpeas import models
+from webtest import TestApp
+from pyramid.paster import get_app
+from scatterpeas.models import User, Alias, Reminder, RRule, Job
+
+TEST_DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql:///test_scatterpeas'
+)
+
+os.environ['DATABASE_URL'] = TEST_DATABASE_URL
+
+# need some global variables
+uuid = ''
+alias_id = ''
+
+
+@pytest.fixture(scope='session')
+def connection(request):
+    engine = create_engine(TEST_DATABASE_URL)
+    models.Base.metadata.create_all(engine)
+    connection = engine.connect()
+    models.DBSession.registry.clear()
+    models.DBSession.configure(bind=connection)
+    models.Base.metadata.bind = engine
+    request.addfinalizer(models.Base.metadata.drop_all)
+    return connection
+
+
+@pytest.fixture()
+def db_session(request, connection):
+    from transaction import abort
+    trans = connection.begin()
+    request.addfinalizer(trans.rollback)
+    request.addfinalizer(abort)
+
+    return models.DBSession
+
+
+@pytest.fixture()
+def app(db_session):
+    app = get_app('ScatterPeas/testing.ini')
+    return TestApp(app)
+
+
+# fails if the homepage doesn't load
+def test_get_home(app):
+    response = app.get('/')
+    assert response.status_code == 200
+    assert 'The pod for all of your scattered peas.' in response.body
+
+
+# fails if we move the create form out of home; mostly a reminder
+# to fix the wiring
+def test_get_create_user_form(app):
+    response = app.get('/')
+    assert response.status_code == 200
+    assert 'id="sign-up-form"' in response.body
+
+
+# this could succeed, fail and return to the homepage with an error
+# message for the user, or just blow up in some unforeseen way.
+def test_create_user(app):
+    global uuid
+    global alias_id
```

Comment:
gah!  Why are you doing this?  This means that you've got some sort of global state that you are mutating with your tests.  Are there other tests that depend on this happening in a particular way?  That's awful!